ehrql/backends/tpp.py

```diff
@@ -99,24 +109,31 @@ def modify_query_variables(self, variables):
         # return it unmodified
         if self.include_t1oo:
             return variables
-        # Otherwise we add an extra condition to the population definition which is that
-        # the patient does *not* appear in the T1OO table.
-        variables = dict(variables)
-        variables["population"] = qm.Function.And(
-            variables["population"],
-            qm.Function.Not(
-                qm.AggregateByPatient.Exists(
-                    # We don't currently expose this table in the user-facing schema. If
-                    # we did then we could avoid defining it inline like this.
-                    qm.SelectPatientTable(
-                        "t1oo",
-                        # It doesn't need any columns: it's just a list of patient IDs
-                        schema=qm.TableSchema(),
-                    )
-                )
-            ),
+        # Otherwise we add an extra condition to the population definition to include
+        # only patients who we know have not opted out
+        patients = ehrql.tables.tpp.patients
+        registrations = ehrql.tables.tpp.practice_registrations
+        extra_condition = (
+            # Include only patients who do *not* have a T1OO recorded
+            ~t1oo.exists_for_patient()
+            & (
+                # Of those, include only those who have a currently active practice
+                # registration (otherwise they may have recorded a T1OO elsewhere) ...
+                registrations.where(
+                    registrations.end_date.is_null()
```

Comment:
Can I verify some assumptions? In the real-world it's only valid for a patient to have one registration at any one time? Does the data reflect that? If the data contains a patient with a registration without an end date, and then a second registration with a later start date than the first registration but *with* an end date, has the first registration ended by default?

I.e. wondering if this check should be "the most recently started registration has a null end date" instead of "there exists at least one registration with a null end date".

Reply:
> In the real-world it's only valid for a patient to have one registration at any one time?

I don't actually know this is the case. It's certainly _normal_ for them to only have one registration at a time, but I don't know that it's completely _invalid_ for them to have multiple.

I think if the data shows an "open" practice registration than we have to take that at face value.